build/configs/imxrt1050-evk/imxrt1050-evk_download.sh

```diff
@@ -115,12 +129,55 @@ cp $FLASH_IMG $IMXRT1050_USB_PATH
 
 elif [[ "${CONFIG_BUILD_PROTECTED}" == "y" ]]; then
 
+echo "Downloading tinyara_flash.bin for Protected build"
 
-sed '$d' $OS_DIR_PATH/../build/output/bin/tinyara.hex > $OS_DIR_PATH/../build/output/bin/tinyara_temp.hex
-cat $OS_DIR_PATH/../build/output/bin/tinyara_temp.hex $OS_DIR_PATH/../build/output/bin/tinyara_user.hex > $OS_DIR_PATH/../build/output/bin/tinyara_prot.hex
-echo "Downloading tinyara_prot.hex for Protected build"
-cp $OS_DIR_PATH/../build/output/bin/tinyara_prot.hex $IMXRT1050_USB_PATH
+# Prepare the binary paths
+
+BIN_PATH=${OS_DIR_PATH}/../build/output/bin
+KERN_IMG=${BIN_PATH}/tinyara.bin
+USER_IMG=${BIN_PATH}/tinyara_user.bin
+FLASH_IMG=${BIN_PATH}/tinyara_flash.bin
+
+# Extract partition sizes form CONFIG_FLASH_PART_SIZE
+OIFS=$IFS
+IFS=","
+NAMES=($CONFIG_FLASH_PART_NAME)
+SIZES=($CONFIG_FLASH_PART_SIZE)
+
+# Set the flash partition sizes.
+for i in "${!NAMES[@]}"; do
```

Comment:
I think we'd better to use part type instead of name.
Kernel and user binary have a type of "kernel" and "bin".